### PR TITLE
*: add unnecessary untracked files to .gitignore and clean up them

### LINKED ISSUE #49070: please add unnecessary untracked files to gitignore and should clean them up in `make clean`
## Enhancement

After I failed to run `make test` or `make ut`, I found that following paths exists and you can see that `git` treats them as untracked files. 

```
Untracked files:
  (use "git add <file>..." to include in what will be committed)
	pkg/domain/extract/
	pkg/executor/test/executor/replayer/
	pkg/server/handler/optimizor/replayer/
	test_coverage/
	tests/integrationtest/replayer/
```

so it is better to add them in `.gitignore` file and to remove them with `make clean`

### PR DESCRIPTION
**Pull request by yoshikipom**
### What problem does this PR solve?
Issue Number: close #49070

Problem Summary: 
After `make test` or `make ut`, some unnecessary files can be shown as untracked files for git. We should prevent accidental commit for them.

### What changed and how does it work?
After this change, files pointed out in #49070 won't be shown in untracked files and you can remove them by executing `make clean`.

There are 3 types of files which are reported in the issue.

#### `./test_coverage`
Since test coverage files should be committed, ignoring this folder makes sense.

#### `**/replayer`
Although 3 folders are reported (`pkg/executor/test/executor/replayer/`, `pkg/server/handler/optimizor/replayer/`, `tests/integrationtest/replayer/`), `replayer` folder can be made in more places. (e.g. `./pkg/server/replayer`, `./pkg/domain/replayer`, ) . I added file names instead of listing folder names.

The file names and the folder name are defined here.
https://github.com/pingcap/tidb/blob/master/pkg/util/replayer/replayer.go#L73-L87

Ignoring `**/replayer` except for `./pkg/util/replayer` can be an option. But I don't choose it because I can't ensure that no one will add another `**/replayer`.

#### `.pkg/domain/extract/`
The folder can be created in https://github.com/pingcap/tidb/blob/master/pkg/domain/extract.go#L493-L494
But I can't see any untracked files in this folder since folder deletion is done in each test case( e.g. https://github.com/pingcap/tidb/blob/master/pkg/domain/extract_test.go#L38).

I added this case in case I missed something.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [x] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -87,8 +87,15 @@ check-parallel:
 		xargs -0 grep -F -n "t.Parallel()" || \
 		! echo "Error: all the go tests should be run in serial."
 
-clean: failpoint-disable
+clean: failpoint-disable clean_tmp_test_data
 	$(GO) clean -i ./...
+	
+clean_tmp_test_data:
```
Comment (hawkingrei) [inline Makefile:93]: It has been fixed by #49132. we can fix it in the tests. so can we remove it?

Comment (yoshikipom) [reply Makefile]: @hawkingrei  Thank you for pointing out. I agree that we should fix the tests. I should have checked the file histories. I'd like to discuss what should be done in this PR.

I treated 3 type of folder here. I understood `3. extract` was completely fixed on the PR you mentioned. I removed  change for `3. extract`. 
6bd091f

1. test_coverage
2. replayer
3. extract

For `1. test_coverage`, there were no changes in the PR. Is it okay to leave my change?

Regarding `2. replayer`, I think some cases were missed in the fix. I still can see some untracked files.  

```
$ git status
On branch master
Your branch is up to date with 'origin/master'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        pkg/domain/replayer/

nothing added to commit but untracked files present (use "git add" to track)

# my master branch contains commits for the fix.
$ git branch --contains a6c6042|grep master 
* master
```

I tried to fix the reamining issues for `replayer` folder. Could you review this additional change? [a85bb6e](https://github.com/pingcap/tidb/pull/49314/commits/a85bb6e7b6c95a808e69dab0a07ee829589a267d) 
`planReplayerTaskDumpWorker.HandleTask` method also call `replayer.GeneratePlanReplayerFile`. We have to care test cases which call `planReplayerTaskDumpWorker.HandleTask`.

File: pkg/server/handler/optimizor/plan_replayer_test.go [hunk 2]
```diff
@@ -112,6 +113,7 @@ func TestDumpPlanReplayerAPI(t *testing.T) {
 	defer server.Close()
 
 	filename, fileNameFromCapture := prepareData4PlanReplayer(t, client, dom)
+	defer os.RemoveAll(replayer.GetPlanReplayerDirName())
```
Comment (hawkingrei) [inline pkg/server/handler/optimizor/plan_replayer_test.go:116]: Please remember to run ```make bazel_prepare```.

Comment (yoshikipom) [reply pkg/server/handler/optimizor/plan_replayer_test.go:116]: @hawkingrei  I did it and pushed the fix on  [df088c1](https://github.com/pingcap/tidb/pull/49314/commits/df088c15cdc26b2a7f57a6d167e0d72800219c64)  (and setup pre-commit by using `.pre-commit-config.yaml`).

**Review by hawkingrei — APPROVED** [review]

**Review by YangKeao — APPROVED** [review]

**Review by lance6716 — APPROVED** [review]

**Pull request merged**